pennylane/transforms/core/transform.py

```diff
@@ -42,6 +46,9 @@ def transform(
 
             * The expand transform must have the same type hinting as a quantum transform.
 
+        qnode_transform (Union[callable, None]): A function that takes a QNode as input and returns another
+            QNode that applies the given transform. If None, a ``TransformProgram`` will be added to
+            the QNode, and the resulting QNode is returned.
```

Comment:
```suggestion
            QNode that applies the given transform. If None, the transform will be added to
            the QNode's :class:`~.TransformProgram`, and the resulting QNode is returned.
```

Reply:
Maybe also mention that the Callable must have the call signature `(qnode, targs, tkwargs)`